gstd/src/sync/mutex.rs

```diff
@@ -30,6 +30,22 @@ use core::{
     task::{Context, Poll},
 };
 
+static mut NEXT_MUTEX_ID: MutexId = MutexId::new();
+
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone, Copy)]
+pub(crate) struct MutexId(u32);
+
+impl MutexId {
+    pub const fn new() -> Self {
+        MutexId(1)
+    }
+
+    pub fn next(self) -> Self {
+        let id = self.0.wrapping_add(1);
+        MutexId(if id == 0 { 1 } else { id })
```

Comment:
?

Reply:
Re-written to make the intent more clear